### PR TITLE
DAT-64906 send gzipped payload from client to server

### PR DESCRIPTION
**Pull request by guyshemer**

### CODE REVIEW

**Review by edengal — DISMISSED** [review]

File: timbermill-java/timbermill-client/src/main/java/com/datorama/oss/timbermill/pipe/TimbermillServerOutputPipe.java [hunk 1]
```diff
@@ -128,9 +132,10 @@ private void sendEvents(EventsWrapper eventsWrapper) throws IOException {
     }
 
     private void sendEventsOverConnection(HttpURLConnection httpCon, byte[] eventsWrapperBytes) throws IOException {
-		try (OutputStream os = httpCon.getOutputStream()) {
-			os.write(eventsWrapperBytes, 0, eventsWrapperBytes.length);
-		}
+        try (OutputStream httpConOS = httpCon.getOutputStream();
```
Comment (guyshemer) [inline timbermill-java/timbermill-client/src/main/java/com/datorama/oss/timbermill/pipe/TimbermillServerOutputPipe.java:135]: support gzip/non-gzip types on client side. mainly for testing purposes for backward-compatibility

File: timbermill-java/timbermill-server/src/main/java/com/datorama/timbermill/server/service/GzippedInputStreamWrapper.java [hunk 2]
```diff
@@ -0,0 +1,161 @@
+package com.datorama.timbermill.server.service;
```
Comment (guyshemer) [inline timbermill-java/timbermill-server/src/main/java/com/datorama/timbermill/server/service/GzippedInputStreamWrapper.java:1]: https://stackoverflow.com/questions/16638345/how-to-decode-gzip-compressed-request-body-in-spring-mvc

File: timbermill-java/timbermill-server/src/main/java/com/datorama/timbermill/server/service/RequestWrapperFilter.java [hunk 3]
```diff
@@ -11,11 +11,26 @@
...
+import kamon.metric.Timer;
+import static com.datorama.oss.timbermill.common.KamonConstants.GZIP_DECOMPRESS_REQUEST_DURATION;
+
 @Component
 public class RequestWrapperFilter extends OncePerRequestFilter {
 
 	@Override
 	protected void doFilterInternal(HttpServletRequest httpServletRequest, HttpServletResponse httpServletResponse, FilterChain filterChain) throws ServletException, IOException {
-		filterChain.doFilter(new ContentCachingRequestWrapper(httpServletRequest), httpServletResponse);
+
```
Comment (guyshemer) [inline timbermill-java/timbermill-server/src/main/java/com/datorama/timbermill/server/service/RequestWrapperFilter.java:24]: decide according to header if it has gzip payload or not , and wrap with appropriate wrapper accordingly

File: timbermill-java/timbermill-server/src/main/java/com/datorama/timbermill/server/service/TimbermillController.java [hunk 4]
```diff
@@ -73,9 +73,9 @@ public String ingestEventsNew(@RequestBody @Valid EventsList events) {
 
 	@ExceptionHandler(HttpMessageNotReadableException.class)
 	@ResponseBody
-	public ResponseEntity<?> handleEmployeeNotFoundException(HttpServletRequest request, Exception ex) throws IOException {
-		ContentCachingRequestWrapper wrapper = (ContentCachingRequestWrapper) request;
-		String body = IOUtils.toString(wrapper.getContentAsByteArray(), wrapper.getCharacterEncoding());
+	public ResponseEntity<?> handleHttpMessageNotReadableException(HttpServletRequest request, Exception ex) throws IOException {
```
Comment (guyshemer) [inline timbermill-java/timbermill-server/src/main/java/com/datorama/timbermill/server/service/TimbermillController.java:76]: method name probably came from other domain question

File: timbermill-java/timbermill-server/src/main/java/com/datorama/timbermill/server/service/TimbermillController.java [hunk 5]
```diff
@@ -73,9 +73,9 @@ public String ingestEventsNew(@RequestBody @Valid EventsList events) {
...
+		HttpServletRequestWrapper wrapper = (HttpServletRequestWrapper) request;
```
Comment (guyshemer) [inline timbermill-java/timbermill-server/src/main/java/com/datorama/timbermill/server/service/TimbermillController.java:77]: using common parent class for both wrapper types

**Review by edengal — APPROVED** [review]

**Pull request merged**